server/controllers/documents.js

```diff
@@ -0,0 +1,204 @@
+  },
+
+  update(req, res) {
+    const query = req.params.id;
+    const loggedInUser = req.user;
+    if (!loggedInUser) {
+      return res.status(403).send({ message: 'Request Denied! SignIn to update documents'})
+    }
+    Documents.findById(query)
+      .then(doc => {
+        if(!doc) {
+          return res.status(404).send({ message: 'Document not found'});
+        };
+
+        const isOwner = checkDocOwner(doc.ownerId, loggedInUser.id);
+        if(!isOwner) {
+          return res.status(401).send({ message: 'Request denied' });
+        };
+        doc.update({
+          title: req.body.title || doc.title,
+          content: req.body.content || doc.content,
+          access: req.body.access || doc.access,
+          ownerId: doc.ownerId
+        })
+          .then(() => 
+            res.status(200).send({ message: 'Document updated successfully'}))
+          .catch(error => 
+            res.status(500).send({ message: 'Server error', error: error}));
+      })
+      .catch(error => res.status(500).send({message: 'Server error'}));
+  },
+
+  deleteOne(req, res) {
+    const loggedInUser = req.user
+    const isAdmin = loggedInUser.roleId === 3;
+    const queryId = req.params.id;
+    ;
+
+    Documents.findById(queryId)
+      .then(doc => {
+        if(!doc) {
+          return res.status(404).send({ message: 'Document does not exist'});
+        };
+        if (isAdmin) {
+          const isPrivate = doc.access === 'private';
+          const ownedByAdmin = doc.ownerId === loggedInUser.id;
+          if(isPrivate && !ownedByAdmin) {
+            return res.status(401).send({ message: 'Request denied'});
+          }
+          return doc.destroy()
+            .then(() => res.status(200)
+              .send({ message: 'Document deleted successfully'}))
+            .catch(error => res.status(500)
+              .send({ message: 'Server error', error: error}));
+        }
+        const isOwner = checkDocOwner(doc.ownerId, loggedInUser.id); 
+        if (!isOwner) {
+          return res.status(401).send({ message: 'Request denied'});
+        }
+        doc.destroy()
+          .then(() => res.status(200)
+            .send({message: 'document deleted successfully'}))
+          .catch(() => res.status(500)
+            .send({ message: 'Server error', error: error}));     
+      })
+      .catch(error => res.status(500)
+        .send( {message: 'Server error', error: error}));  
+  },
+
+  search(req, res) {
+    const { docTitle } = req.query;
+    const isAdmin = req.user.roleId === 3;
+    const loggedInUser = req.user;
+
+    Documents.findAll({
+      where: {
+        title: docTitle
+      }
+    })
+      .then(docs => {
+        if (!docs) {
+          return res.status(404).send({ message: 'Not Found'});
+        };
+
+        const isWriter = checkUserRole(loggedInUser.roleId);
+        if (isWriter) {
+          const writerDocs = docs.filter(doc => {
```

Comment:
The checks here are not that clear to me. I bet when you came back to this code after few weeks, you'll scratch your head to remember as well.